jac/support/jac-lang.org/docs/learn/tutorial/2_building-a-rag-chatbot.md

```diff
@@ -1,37 +1,42 @@
 # Building a RAG Chatbot with Jac Cloud and Streamlit (Part 2/3)
 
-Now that we have a jac application served up, let's build a simple chatbot using Retrieval Augmented Generation (RAG) with Jac Cloud and Streamlit as our frontend interface.
+Now that we have a jac application served up, In this section, we'll build a simple chatbot using **Retrieval Augmented Generation (RAG)** with **Jac Cloud** for the backend and **Streamlit** for the frontend interface.
 
 ### Preparation / Installation
 
-There are a couple of additional dependenices we need here
+Before we start, let's install a few additional dependencies. Run the following command in your terminal:
 
 ```bash
 pip install mtllm==0.3.2 jac-streamlit==0.0.3 langchain==0.1.16 langchain_community==0.0.34 chromadb==0.5.0 pypdf==4.2.0
 ```
 
+- `mtllm`: A toolkit for multi-task learning with language models.
+- `jac-streamlit`: A plugin that integrates jaclang with Streamlit.
+- `langchain`, `langchain_community`: Essential tools for handling the language generation logic in your chatbot.
+- `chromadb`: A vector database to store and retrieve embeddings.
+- `pypdf`: A Python library to handle PDF documents.
+
+
 ## Building a Streamlit Interface
 
 Before we begin building out our chatbot, let's first build a simple GUI to interact with the chatbot. Streamlit offers several Chat elements, enabling you to build Graphical User Interfaces (GUIs) for conversational agents or chatbots. Leveraging session state along with these elements allows you to construct anything from a basic chatbot to a more advanced, ChatGPT-like experience using purely Python code.
 
 Luckily for us, jaclang has a plugin for streamlit that allows us to build web applications with streamlit using jaclang. In this part of the tutorial, we will build a frontend for our conversational agent using streamlit. You can find more information about the `jac-streamlit` plugin [here](https://github.com/Jaseci-Labs/jaclang/blob/main/support/plugins/streamlit/README.md).
 
-First, let's create a new file called `client.jac`. This file will contain the code for the frontend chat interface.
+Begin by creating a file named `client.jac`. This file will define the frontend logic for interacting with users and displaying responses from the chatbot.
```

Comment:
Change to "for interacting with the chatbot" or "for users to interact with the chatbot"